src/rust/src/x509.rs

```diff
@@ -55,7 +92,100 @@ fn parse_x509_extension(
     let oid = asn1::ObjectIdentifier::from_der(der_oid).unwrap();
 
     let x509_module = py.import("cryptography.x509")?;
-    if oid == *TLS_FEATURE_OID {
+    if oid == *SUBJECT_ALTERNATIVE_NAME_OID {
+        let sans = pyo3::types::PyList::empty(py);
+        for san in asn1::parse_single::<asn1::SequenceOf<GeneralName>>(ext_data)? {
+            let pyobj = match san {
+                GeneralName::OtherName(data) => {
+                    let oid = x509_module
+                        .call_method1("ObjectIdentifier", (data.type_id.to_string(),))?
+                        .to_object(py);
+                    x509_module
+                        .call_method1("OtherName", (oid, data.value.data()))?
+                        .to_object(py)
+                }
+                GeneralName::RFC822Name(data) => x509_module
+                    .getattr("RFC822Name")?
+                    .call_method1("_init_without_validation", (data.0,))?
+                    .to_object(py),
+                GeneralName::DNSName(data) => x509_module
+                    .getattr("DNSName")?
+                    .call_method1("_init_without_validation", (data.0,))?
+                    .to_object(py),
+                GeneralName::DirectoryName(data) => {
+                    let py_list = pyo3::types::PyList::empty(py);
+                    for rdn in data {
+                        let py_set = pyo3::types::PySet::empty(py)?;
+                        for attribute in rdn {
+                            let oid = x509_module
+                                .call_method1("ObjectIdentifier", (attribute.type_id.to_string(),))?
+                                .to_object(py);
+                            let tag_enum = py
+                                .import("cryptography.x509.name")?
+                                .getattr("_ASN1_TYPE_TO_ENUM")?;
+                            let py_tag = tag_enum.get_item(attribute.value.tag().to_object(py))?;
+                            // TODO: is it safe to assume utf8? Current Python does.
```

Comment:
Pretty much, https://datatracker.ietf.org/doc/html/rfc4514#section-2.4

As you all know, DN's are uncommon and the rare cases follow LDAP syntax. LDAP implementations use UTF-8 in RDN attribute values.